internal/testutility/utility.go

```diff
@@ -0,0 +1,63 @@
+package testutility
+
+import (
+	"encoding/json"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/kr/pretty"
+)
+
+// LoadHelper loads a JSON fixture file and returns the decoded version.
+func LoadHelper[V any](t *testing.T, path string) V {
```

Comment:
nit: `Helper` isn't too helpful of a name :) 

Maybe `LoadJSONFixture` instead ?